src/components/signup/SignupState.js

```diff
@@ -144,6 +205,13 @@ const Signup = ({ navigation, screenProps }: { navigation: any, screenProps: any
 
     log.info('Sending new user data', state)
     try {
+      if (state.w3Token) {
+        await API.checkWeb3Email({
```

Comment:
we could do this earlier in the "gotophone" section
and run it in background without await.
API.checkweb...catch

Reply:
I thought that this verification should be right before saving a user. So after user press submit button - there is a verification of email used by client compare email provided by web3

Reply:
it doesnt really matter, the isEmailVerified is saved to the user record on the server. so it can happen in the background a long time before saving the user ie it doesnt have to be immediatly before saving user.